src/category_theory/idempotents/biproducts.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import category_theory.idempotents.karoubi
+import category_theory.additive.basic
+
+/-!
+
+# Biproducts in the idempotent completion of a preadditive category
+
+In this file, we define an instance expressing that if `C` is an additive category,
+then `karoubi C` is also an additive category.
+
+We also obtain that for all `P : karoubi C` where `C` is a preadditive category `C`, there
+is a canonical isomorphism `P ⊞ P.supplement ≅ (to_karoubi C).obj P.X` in the category
+`karoubi C` where `P.supplement` is the formal direct factor of `P.X` corresponding to
+the idempotent endomorphism `𝟙 P.X - P.p`.
+
+-/
+
+noncomputable theory
+
+open category_theory.category
+open category_theory.limits
+open category_theory.preadditive
+
+universes v
+
+namespace category_theory
+
+namespace idempotents
+
+namespace karoubi
+
+variables {C : Type*} [category.{v} C] [preadditive C]
+
+namespace biproducts
+
+/-- The `bicone` used in order to obtain the existence of
+the biproduct of a functor `J ⥤ karoubi C` when `C`. -/
```

Comment:
Sentence ends before it should?

Reply:
Thanks. I have fixed this. (And also changed `supplement` into `complement`.)